src/script/interpreter.cpp

```diff
@@ -1681,6 +1752,205 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
+                    const CTxIn inp = inps->at(idx);
+                    const CTxOut spent_utxo = spent_outputs[idx];
+
+                    switch (opcode)
+                    {
+                        case OP_INSPECTINPUTOUTPOINT:
+                        {
+                            valtype vchPrevTxid;
+                            vchPrevTxid.insert(vchPrevTxid.begin(), inp.prevout.hash.begin(), inp.prevout.hash.begin() + 32);
+                            stack.push_back(vchPrevTxid);
+                            push4_le(stack, inp.prevout.n);
+
+                            // Push the outpoint flag
+                            valtype vchOutpointFlag(1);
+                            vchOutpointFlag[0] = GetOutpointFlag(inp);
+                            stack.push_back(vchOutpointFlag);
+                            break;
+                        }
+                        case OP_INSPECTINPUTASSET:
+                        {
+                            pushasset(stack, spent_utxo.nAsset);
+                            break;
+                        }
+                        case OP_INSPECTINPUTVALUE:
+                        {
+                            pushvalue(stack, spent_utxo.nValue);
+                            break;
+                        }
+                        case OP_INSPECTINPUTSCRIPTPUBKEY:
+                        {
+                            valtype vchScriptPubKeySha256;
+                            vchScriptPubKeySha256.insert(vchScriptPubKeySha256.begin(), cache->m_spent_output_spk_single_hashes[idx].begin(), cache->m_spent_output_spk_single_hashes[idx].begin() + 32);
+                            pushspk(stack, spent_utxo.scriptPubKey, vchScriptPubKeySha256);
+                            break;
+                        }
+                        case OP_INSPECTINPUTSEQUENCE:
+                        {
+                            push4_le(stack, inp.nSequence);
+                            break;
+                        }
+                        case OP_INSPECTINPUTISSUANCE:
+                        {
+                            if (!inp.assetIssuance.IsNull()) {
+                                valtype vchAssetBlindingNonce, vchAssetEntropy;
+                                pushvalue(stack, inp.assetIssuance.nInflationKeys);
+                                pushvalue(stack, inp.assetIssuance.nAmount);
+                                // Next push Asset entropy
+                                vchAssetEntropy.insert(vchAssetEntropy.begin(), inp.assetIssuance.assetEntropy.begin(), inp.assetIssuance.assetEntropy.end());
+                                stack.push_back(vchAssetEntropy);
+                                // Finally push blinding nonce
+                                // By pushing the this order, we make sure that the stack top is empty
+                                // iff there is no issuance. Note that nInflationKeys can be null and can push false
```

Comment:
Either one of inflation keys or the asset amount could be Null.

Reply:
Updated, it was my misunderstanding that these outputs should be non-zero.